scipy/spatial/ckdtree.pyi

```diff
@@ -1,9 +1,221 @@
+import sys
+from typing import (
+    Any,
+    Dict,
+    Generic,
+    List,
+    Optional,
+    overload,
+    Set,
+    Tuple,
+    TypeVar,
+    Union,
+)
+
 import numpy as np
+import numpy.typing as npt
+from scipy.sparse import coo_matrix, dok_matrix
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+# TODO: Replace `ndarray` with a 1D float64 array when possible
+_BoxType = TypeVar("_BoxType", None, np.ndarray)
+
+# Copied from `numpy.typing._scalar_like._ScalarLike`
+# TODO: Expand with 0D arrays once we have shape support
+_ArrayLike0D = Union[
+    bool,
+    int,
+    float,
+    complex,
+    str,
+    bytes,
+    np.generic,
+]
+_WeightType = Union[npt.ArrayLike, Tuple[npt.ArrayLike, npt.ArrayLike]]
 
 class cKDTreeNode:
     @property
     def data_points(self) -> np.ndarray: ...
     @property
     def indices(self) -> np.ndarray: ...
 
-class cKDTree: ...
+    # Read-only attributes that are technically not properties
```

Comment:
Is this meant to be "are technically properties"?

Reply:
You mean like it's technically a descriptor, not actually an instance of `property`?

Reply:
I like it, done as of https://github.com/scipy/scipy/pull/14024/commits/6c43b33d2af1b955a6be27f08fe1a496c3976e13.